File: content/snippets/python/language_basics/classes_and_objects.py
#!/usr/bin/env python3
from dataclasses import dataclass


@dataclass
class Person:
    name: str
    age: int

    def describe(self) -> str:
        return f"{self.name} is {self.age} years old"

    def birthday(self) -> None:
        self.age += 1


def main() -> None:
    person = Person("Alice", 29)
    print(person.describe())
    person.birthday()
    print(person.describe())


if __name__ == "__main__":
    main()

File: content/snippets/go/language_basics/classes_and_objects.go
package main

import "fmt"

type Person struct {
	Name string
	Age  int
}

func (person Person) Describe() string {
	return fmt.Sprintf("%s is %d years old", person.Name, person.Age)
}

func (person *Person) Birthday() {
	person.Age++
}

func main() {
	person := Person{Name: "Alice", Age: 29}
	fmt.Println(person.Describe())
	person.Birthday()
	fmt.Println(person.Describe())
}
